libgadget/utils/system.c

```diff
@@ -453,6 +528,50 @@ get_physmem_bytes()
     return 64 * 1024 * 1024;
 }
 
+int
+_MPIU_Barrier(const char * fn, const int line, MPI_Comm comm)
+{
+    int ThisTask, NTask;
+    MPI_Comm_size(comm, &NTask);
+    MPI_Comm_rank(comm, &ThisTask);
+    int * recvbuf = malloc(sizeof(int) * NTask);
```

Comment:
system malloc? Should this be ta_malloc?

Reply:
No it shouldn't because  system.c sits lower on the stack

Reply:
We use mymalloc and ta_malloc elsewhere in system.c

Reply:
I am surprised. We probably wanted to minimize fragmentation since these routines are called frequently.  I changed it to use ta_malloc.

Reply:
Yes, I think that was it. Thanks.